### PR TITLE
Computed properties Python Support

### PR DESCRIPTION
**Pull request by bambriz**
# Description
Adds support for computed properties.

Sample Usage:
```python
 computed_properties = [{'name': "cp_str_len", 'query': "SELECT VALUE LENGTH(c.stringProperty) FROM c"}]
container = db.create_container("Container_id", partition_key= PartitionKey(path="\pk"), computed_properties=computed_properties)
queried_items = list(container.query_items(query='Select * from c Where c.cp_str_len = 9', partition_key="test")"
```

Also see [Feature Doc](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/query/computed-properties?tabs=dotnet)

Computed properties cannot have aggregate functions. 

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### 4.5.2b4 (Unreleased)
 
 #### Features Added
+* Added conditional support for preview feature of Computed Properties on Python SDK. See [PR 33626](https://github.com/Azure/azure-sdk-for-python/pull/33626)
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:6]: What's it conditional on? Or does this just mean it's still a service preview feature?
Does the service plan to GA this feature before the next SDK GA?

Comment (bambriz) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:6]: This is supposed to mean that it is still a preview feature that needs to be enabled on the account. As of now, I do not know if they plan to GA this feature before the next SDK GA.

Comment (simorenoh) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:6]: we can just rename this to match other feature entries that are in preview like priority based throttling:
image

**Review by simorenoh — COMMENTED** [review]
awesome stuff, thanks Bryan! let's make sure to update the changelog and it should be good to go 👍

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (bambriz) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: I went ahead and changed the wording of the changelog, but I will omit the feature doc link in the changelog, it seems the pipeline doesn't like that link and labels it as broken. I'll include it instead in the PR description.

File: sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py [hunk 2]
```diff
@@ -175,6 +175,12 @@ async def create_container(
             has changed, and act according to the condition specified by the `match_condition` parameter.
         :keyword match_condition: The match condition to use upon the etag.
         :paramtype match_condition: ~azure.core.MatchConditions
+        :keyword List[Dict[str, str]] computed_properties: Computed properties must be at the top level in the item and
+            can't have a nested path. Each computed property definition has two components: a name and a query.
+            The name is the computed property name, and the query defines logic to calculate the property value
+            for each item. Computed properties are scoped to an individual item and therefore can't use values
+            from multiple items or rely on other computed properties. Every container can have
+            a maximum of 20 computed properties.
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py:183]: didn't see this before but this is way too much information for the kwarg - look at the Java doc for this functionality for instance:
image

**Review by simorenoh — COMMENTED** [review]
let's shorten the information in the docs and we should be good, thanks!

File: sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py (hunk 2 above)
Comment (bambriz) [reply sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py]: Thank you for catching this. I have updated them to match the Java docstrings more.

**Review by simorenoh — APPROVED** [review]

**Pull request merged**